markdown/bitburner.ns.settailfontsize.md

```diff
@@ -0,0 +1,37 @@
+<!-- Do not edit this file. It is automatically generated by API Documenter. -->
+
+[Home](./index.md) &gt; [bitburner](./bitburner.md) &gt; [NS](./bitburner.ns.md) &gt; [setTailFontSize](./bitburner.ns.settailfontsize.md)
+
+## NS.setTailFontSize() method
+
+Set the font size of the tail window of a script.
+
+**Signature:**
+
+```typescript
+setTailFontSize(pixel?: number, pid?: number): void;
+```
+
+## Parameters
+
+|  Parameter | Type | Description |
+|  --- | --- | --- |
+|  pixel | number | _(Optional)_ Optional. The new font size in pixels. If omitted, the default tail font size is used. |
+|  pid | number | _(Optional)_ Optional. PID of the script having its tail closed. If omitted, the current script is used. |
+
+**Returns:**
+
+void
+
+## Remarks
+
+RAM cost: 0 GB
+
+This overwrites the tail font size and forces an update of the tail window's contents.
+
+The font size is saved across restarts.
```

Comment:
This has a variety of implications that I'm not so sure about.

For one thing, `temporary` scripts won't be saved (although I think they also don't have logs, or at least you don't tend to look at them.)

In the more general sense, saving this as a property in this way, particularly as a *persistent* property, means that changing the font size in settings won't do anything (because the font size is already specified on all the scripts).

Also also, we already have tailProps, which are non-persisted. This feels like something that belongs there?

Reply:
I can certainly move the property to the tailProps.
At first I thought it would make more sense to have that stored, since this will mostly come in handy for custom UIs that are just that tiny bit too big and don't fit in the width and height set.
Those properties are lost, so that logic doesn't work out.

That also solves the temporary scripts stuff.
However, the style editor changing the font size is a bit more tricky:
Quickly changing the tail font size should probably stay.
My first idea was to just make the tail font size typed as `number | undefined` and only set it if the font size has been altered. Then the log box will use the font size `fontSize: runningScriptObj.tailProps.fontSize ?? Settings.styles.tailFontSize`.
This way any scripts that don't have custom font sizes will use the default set by the style editor and show the changes, but custom UIs don't get affected and possibly messed up. It wouldn't make sense to scale those as well.